theano/compile/tests/test_function_module.py

```diff
@@ -239,6 +239,37 @@ def test_copy(self):
         f(1, 2)  # put them out of sync
         self.assertFalse(f(1, 2) == g(1, 2))  # they should not be equal anymore.
 
+    def test_copy_share_memory(self):
+        x = T.fscalar('x')
+        y = T.tanh((x+2)/(x-0.2)**2)
+
+        # test for PerformaLinker, will cover VM_linker later
+        ori = theano.function([x], [y], mode="FAST_COMPILE")
+        cpy = ori.copy(share_memory=True)
+
+        # test if memories shared
+        storage_map_ori = ori.fn.storage_map
+        storage_map_cpy = cpy.fn.storage_map
+        fgraph_ori = ori.maker.fgraph
+        fgraph_cpy = cpy.maker.fgraph
+
+        # assert intermediate and Constants storages are shared
+        i_o_variables = fgraph_cpy.inputs + fgraph_cpy.outputs
+        ori_storages = storage_map_ori.values()
+        for key in storage_map_cpy.keys():
+            if key not in i_o_variables or isinstance(key, theano.tensor.Constant):
```

Comment:
I think that inputs and outputs should have different storage except for Constant and SharedVariable. Why do you think otherwise?